### PR TITLE
AB string

### PR DESCRIPTION
**Pull request by ranshu1601**

### CODE REVIEW

File: anshu/abdulBari/string/compareStr.cpp
Comment (noiceee) [inline anshu/abdulBari/string/compareStr.cpp:1]: What is this function trying to achieve?

File: anshu/abdulBari/string/countVowels.cpp
Comment (noiceee) [inline anshu/abdulBari/string/countVowels.cpp:1]: This code needs to be cleaned, do not put so many conditions inside the if statement, instead create an array containing the vowels, and then check if your element is in the array or not, and proceed accordingly. It is a much clear practice.

File: anshu/abdulBari/string/findDuplicate.cpp
Comment (noiceee) [inline anshu/abdulBari/string/findDuplicate.cpp:1]: Do not give outputs from the function itself, instead return an array from the function, and print the elements using a display function in your main function

File: anshu/abdulBari/string/plaindrome.cpp
Comment (noiceee) [inline anshu/abdulBari/string/plaindrome.cpp:1]: Same here, your function should be generic, and not write to the output stream, instead it should return a value, and using that value, the desired output will be generated in the main function. Also, your output should mention if it's a palindrome or not, and not simple true false

File: anshu/abdulBari/string/reverseWord.cpp
Comment (noiceee) [inline anshu/abdulBari/string/reverseWord.cpp:1]: Same here, you should avoid void functions, and only use them if you are modifying the contents of an object permanently.

File: checks.json
Comment (noiceee) [inline checks.json:1]: Add this file in gitignore, baar baar ajata hai

File: anshu/abdulBari/string/validStr.cpp
Comment (noiceee) [inline anshu/abdulBari/string/validStr.cpp:1]: Take input from the user, and give better outputs stating the answer and not 0 or 1

File: anshu/abdulBari/string/compareStr.cpp
Comment (ranshu1601) [reply anshu/abdulBari/string/compareStr.cpp:1]: this function is comparing two strings and printing the common substring .

File: anshu/abdulBari/string/anagram.cpp
Comment (noiceee) [inline anshu/abdulBari/string/anagram.cpp:1]: For all code, input should be taken from the user, and functions should return a value, using which the output will be generated in the main function. The output should be self-explanatory and not like 0 1 and all.

**Review by noiceee — CHANGES_REQUESTED** [review]

File: anshu/abdulBari/string/countVowels.cpp
Comment (ranshu1601) [reply anshu/abdulBari/string/countVowels.cpp:1]: okay , as of now just merge .

File: checks.json
Comment (ranshu1601) [reply checks.json:1]: just check now

File: anshu/abdulBari/string/validStr.cpp
Comment (ranshu1601) [reply anshu/abdulBari/string/validStr.cpp:1]: Will keep that in mind , tq

File: anshu/abdulBari/string/countVowels.cpp
Comment (noiceee) [reply anshu/abdulBari/string/countVowels.cpp:1]: Ee wala krdo

**Comment by noiceee** [comment]
Please do that vowel one with array

File: anshu/abdulBari/string/countVowels.cpp
Comment (ranshu1601) [reply anshu/abdulBari/string/countVowels.cpp:1]: krta

**Pull request merged**